### PR TITLE
Changed what sequence attributes are included in diff/generate changelog for Snowflake

### PR DESCRIPTION
**Pull request by yodzhubeiskyi**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

Snowflake only supports operating 'startWith' and 'incrementBy' sequence attributes. The rests are set by default and are not allowed to be set or changed. In order for Liquibase to correctly work with those 'fixed' attributes we need to:

- Throw a validation error when the 'fixed' attributes are present in changesets that are being applied to Snowflake
- Include the values in JSON snapshots so the database is accurately and completely represented BUT
- Exclude the 'fixed' attributes from any changesets generated by diffChangelog or generateChangelog so that we don't generate invalid changesets

### CODE REVIEW

**Review by nvoxland — APPROVED** [review]

File: liquibase-snowflake/src/main/java/liquibase/diff/output/changelog/core/MissingSequenceChangeGeneratorSnowflake.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        return priority;
+    }
+
+    @Override
+    public Change[] fixMissing(DatabaseObject missingObject, DiffOutputControl control, Database referenceDatabase, Database comparisonDatabase, ChangeGeneratorChain chain) {
+        Sequence sequence = (Sequence) missingObject;
+
+        CreateSequenceChange change = new CreateSequenceChange();
+        change.setSequenceName(sequence.getName());
+        if (control.getIncludeSchema()) {
```
Comment (KushnirykOleh) [inline liquibase-snowflake/src/main/java/liquibase/diff/output/changelog/core/MissingSequenceChangeGeneratorSnowflake.java:29]: i would rather treat catalogs in DiffOutputControl like in parent class

       ``` if (control.getIncludeCatalog()) {
            change.setCatalogName(sequence.getSchema().getCatalogName());
        }```

**Review by KushnirykOleh — CHANGES_REQUESTED** [review]
one ask for a change, other than that looks good

**Review by KushnirykOleh — APPROVED** [review]

File: liquibase-snowflake/src/main/java/liquibase/diff/output/changelog/core/ChangedSequenceChangeGeneratorSnowflake.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import liquibase.diff.ObjectDifferences;
+import liquibase.diff.output.DiffOutputControl;
+import liquibase.diff.output.changelog.ChangeGeneratorChain;
+import liquibase.structure.DatabaseObject;
+import liquibase.structure.core.Sequence;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ChangedSequenceChangeGeneratorSnowflake extends ChangedSequenceChangeGenerator{
```
Comment (suryaaki2) [inline liquibase-snowflake/src/main/java/liquibase/diff/output/changelog/core/ChangedSequenceChangeGeneratorSnowflake.java:16]: Just wondering if the name of the class should be be SnowflakeChangedSequenceChangeGenerator? Can you please check if there is a pattern, if not am fine..

File: liquibase-snowflake/src/main/java/liquibase/diff/output/changelog/core/MissingSequenceChangeGeneratorSnowflake.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import liquibase.change.Change;
+import liquibase.change.core.CreateSequenceChange;
+import liquibase.database.Database;
+import liquibase.database.core.SnowflakeDatabase;
+import liquibase.diff.output.DiffOutputControl;
+import liquibase.diff.output.changelog.ChangeGeneratorChain;
+import liquibase.structure.DatabaseObject;
+import liquibase.structure.core.Sequence;
+
+public class MissingSequenceChangeGeneratorSnowflake extends MissingSequenceChangeGenerator{
```
Comment (suryaaki2) [inline liquibase-snowflake/src/main/java/liquibase/diff/output/changelog/core/MissingSequenceChangeGeneratorSnowflake.java:12]: Same comment as above

File: liquibase-snowflake/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGeneratorSnowflake.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import liquibase.database.Database;
+import liquibase.database.core.*;
+import liquibase.exception.ValidationErrors;
+import liquibase.sql.Sql;
+import liquibase.sql.UnparsedSql;
+import liquibase.sqlgenerator.SqlGeneratorChain;
+import liquibase.statement.core.CreateSequenceStatement;
+
+public class CreateSequenceGeneratorSnowflake extends CreateSequenceGenerator{
```
Comment (suryaaki2) [inline liquibase-snowflake/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGeneratorSnowflake.java:11]: same comment

**Review by suryaaki2 — CHANGES_REQUESTED** [review]

**Review by suryaaki2 — APPROVED** [review]

File: liquibase-snowflake/src/main/java/liquibase/diff/output/changelog/core/ChangedSequenceChangeGeneratorSnowflake.java (hunk 2 above)
Comment (yodzhubeiskyi) [reply liquibase-snowflake/src/main/java/liquibase/diff/output/changelog/core/ChangedSequenceChangeGeneratorSnowflake.java:16]: The naming here goes according to the rest of the classes in Snowflake extension, so I'd consider it to be correct

**Pull request merged**